addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/handler/NibeHeatPumpHandler.java

```diff
@@ -0,0 +1,544 @@
+            clearCache(coilAddress);
+            return;
+        }
+
+        if (!configuration.enableWriteCommands) {
+            logger.info(
+                    "All write commands denied, ignoring command! Change Nibe heat pump binding configuration if you want to enable write commands.");
+            return;
+        }
+
+        if (!itemsToEnableWrite.contains(coilAddress)) {
+            logger.info(
+                    "Write commands to register '{}' not allowed, ignoring command! Add this register to Nibe heat pump binding configuration if you want to enable write commands.",
+                    coilAddress);
+            return;
+        }
+
+        if (connector != null) {
+
+            VariableInformation variableInfo = VariableInformation.getVariableInfo(pumpModel, coilAddress);
+
+            logger.debug("Usig variable information for register {}: {}", coilAddress, variableInfo);
+
+            if (variableInfo != null && variableInfo.type == VariableInformation.Type.SETTING) {
+
+                int value = convertStateToNibeValue(command);
+                value = value * variableInfo.factor;
+
+                ModbusWriteRequestMessage msg = new ModbusWriteRequestMessage.MessageBuilder().coilAddress(coilAddress)
+                        .value(value).build();
+
+                try {
+                    writeResult = sendMessageToNibe(msg);
+                    ModbusWriteResponseMessage result = (ModbusWriteResponseMessage) writeResult.get(timeout,
+                            TimeUnit.MILLISECONDS);
+                    if (result != null) {
+                        if (result.isSuccessfull()) {
+                            logger.debug("Write message sending to heat pump succeeded");
+
+                        } else {
+                            logger.error("Message sending to heat pump failed, value not accepted by the heat pump");
+                        }
+                    } else {
+                        logger.debug("Something weird happen, result for write command is null");
+                    }
+                } catch (TimeoutException e) {
+                    logger.error("Message sending to heat pump failed, no response");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "No response received from the heat pump");
+                } catch (InterruptedException e) {
+                    logger.error("Message sending to heat pump failed, sending interrupted");
+                } catch (NibeHeatPumpException e) {
+                    logger.error("Message sending to heat pump failed, exception {}", e.getMessage());
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                } catch (ExecutionException e) {
+                    logger.error("Message sending to heat pump failed, exception {}", e.getMessage());
```

Comment:
Assuming this is a bug situation that should be reported, also add `e` as a parameter, so that the logging includes the stacktrace.

Reply:
Done